presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderWindows.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.sql.planner.PlanNodeIdAllocator;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.SymbolAllocator;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.WindowNode;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.PriorityQueue;
+
+public class ReorderWindows
```

Comment:
How you want me to switch only two at a time?
Imagine case:
window nodes with partition by clauses as follows: C,A,D,B
In first iteration I switch C with A, resulting in A,C,D,B
then in next iteration I look at next pair of window nodes: C with D - no switch
then D and B - switch.
We end up with: A,C,B,D which is not correct.
This would be simple and naive approach (which is also incorrect).

I don't think implementing (proper) bubble sort here would make it simpler than using priority queue which underneath implements presumably quicksort or heap sort.